src/main/java/com/emc/mongoose/metrics/util/MetricCollector.java

```diff
@@ -0,0 +1,133 @@
+package com.emc.mongoose.metrics.util;
+
+import io.prometheus.client.Collector;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.LongAdder;
+
+/**
+ @author veronika K. on 09.10.18 */
+public class MetricCollector
+	extends Collector {
+
+	private double loQuantileValue = 0.25;
+	private double hiQuantileValue = 0.75;
+	private final Histogram histogram;
+	private String metricName = String.valueOf(this.hashCode());
+	private final List<String> labelValues = new ArrayList<>();
+	private final List<String> labelNames = new ArrayList<>();
+	private final LongAdder sum = new LongAdder();
+	private final AtomicLong min = new AtomicLong(Long.MAX_VALUE);
+	private final AtomicLong max = new AtomicLong(Long.MIN_VALUE);
+	//
+	//TODO: public getters ?
+
+	public MetricCollector(final int reservoirSize) {
+		this.histogram = new Histogram(reservoirSize);
+	}
+
+	public MetricCollector() {
+		this.histogram = new Histogram();
+	}
+
+	public void update(final int value) {
+		update((long) value);
+	}
+
+	public void update(final long value) {
+		histogram.update(value);
+		sum.add(value);
+		if(value < min.get()) {
+			min.set(value);
+		}
+		if(value > max.get()) {
+			max.set(value);
+		}
+	}
+
+	public MetricCollector name(final String name) {
+		metricName = name + "_" + hashCode();
+		return this;
+	}
+
+	public MetricCollector loQ(final double q) {
```

Comment:
looks to be unneccessary
I think this should be in some kind of builder like a method "addQuantile(final double quantile)" which appends the given quantile (0.25, 0.5, 0.75, ...) to a list of quantiles to calculate